src/SmartComponents/SystemAdvisor/SystemAdvisor.js

```diff
@@ -166,25 +167,43 @@ const BaseSystemAdvisor = () => {
     collapseRows[rowId] = { ...collapseRows[rowId], isOpen };
     setRows(collapseRows);
   };
+  const location = useLocation().pathname?.split('/');
 
   const buildRows = (
     activeReports,
     kbaDetails,
     filters,
     rows,
     searchValue = '',
-    kbaLoading = false
+    kbaLoading = false,
+    isFirstLoad = false
   ) => {
-    const builtRows = activeReports.flatMap((value, key) => {
+    const url = window.location.href;
+    let newActiveReportsList = activeReports;
+    let isRulePresent = url.indexOf('activeRule') ? true : false;
+    if (isRulePresent) {
+      let activeRule = location[2];
```

Comment:
I think with `useParams` you can use `params.id` or something like that, since the route for this is defined as `"/recommendations/:id/:inventoryId/"`.

Reply:
useParams returns undefined , restructured or not. Im not sure if this is because its systemAdvisor is going through redux and federated modules, but originally when making this I tried something like 
  ```
const { id } = useParams();
  const { inventoryId } = useParams();
  or even  let params = useParams();
```
Trying again now im getting the same thing. I get undefined every time, that was originally why I opted for using useLocation.